File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    stack = []
    removeCnt = k
    
    for i, v in enumerate(number):
        while removeCnt > 0 and len(stack) > 0 and stack[-1] < v:
            stack.pop()
            removeCnt -= 1
        
        stack.append(v)
        
    
    while removeCnt > 0:
        stack.pop()
        removeCnt -= 1
        
    return "".join(stack)

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
    const stack = [];
    let removeCnt = k;
    
    [...number].forEach((v, i) => {
        while(removeCnt > 0 && stack.length > 0 && stack[stack.length - 1] < v){
            stack.pop();
            removeCnt--;
        }
        
        stack.push(v);
    })
    
    while(removeCnt > 0){
        stack.pop();
        removeCnt--;
    }
    
    return stack.join("");
}